packages/seal-loupe-adapter/src/LoupeHelper.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Schranz Search package.
+ *
+ * (c) Alexander Schranz <alexander@sulu.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Schranz\Search\SEAL\Adapter\Loupe;
+
+use Loupe\Loupe\Configuration;
+use Loupe\Loupe\Loupe;
+use Loupe\Loupe\LoupeFactory;
+use Schranz\Search\SEAL\Schema\Index;
+
+/**
+ * @experimental
+ */
+final class LoupeHelper
```

Comment:
@toflar What is a little bit different to other search engines is that at creating the `Loupe` instance (what in other cases is the `client) the whole configuration searchablefields, filterablefields, .. is required.

For this I created this Helper class which could also be called `LoupeManager`. It even `serialize` the configuration into a file this way it is more strict as changing updating the configuration can only be done like in other engine via the SchemaManager::createIndex method and so it is more consistent through the whole library and so without calling that the changes apply no effects.

Personally I think a `LoupeManager` for my usecase could look like this:

```php
class LoupeManager
{
    /**
     * @var Loupe[]
     */
    private array $loupe = [];

    public function __construct(
        private readonly LoupeFactory $loupeFactory,
        private string $directory,
    ) {
    }
    
    public function getLoupe(string $index): Loupe
    {
        if (!isset($this->loupe[$index->name])) {
            $configurationFile = $this->directory . $index . '.loup';

            if (!\file_exists($configurationFile)) {
                throw new \LogicException('Index need to be created before accessing Loupe instance.');
            }

            /** @var string $configurationContent */
            $configurationContent = \file_get_contents($configurationFile);

            /** @var Configuration $configuration */
            $configuration = \unserialize($configurationContent);
            
            $this->loupe[$index->name] = $this->loupeFactory->create($this->directory . $index . '.db', $configuration);
        }

        return $this->loupe[$index->name];
    }
    
    public function existIndex(string $index): bool
    {
        return file_exists(($this->directory . $index . '.db', $this->directory . $alias . '.db');
    }
    
    public function dropIndex(string $index): void
    {
         unlink($this->directory . $index . '.db', $this->directory . $alias . '.db');
         unlink($this->directory . $index . '.loup', $this->directory . $alias . '.loup');
    }
    
    public function createIndex(string $index, Configuration $configuration): void
    {
        $configurationContent = \serialize($configuration);
        
        file_put_contents($this->directory . $index . '.loup', $this->directory . $alias . '.loup', $configurationContent);
        touch($this->directory . $index . '.db');
    }
    
    public function alias(string $index, string $alias): void
    {
         // aliases via symlinks for zerodowntime reindexing
         symlink($this->directory . $index . '.db', $this->directory . $alias . '.db');
         symlink($this->directory . $index . '.loup', $this->directory . $alias . '.loup');
    }
    
    public function removeAlias(string $alias): void
    {
         unlink($this->directory . $alias . '.db');
         unlink($this->directory . $alias . '.loup');
    }
}
```

Reply:
Answered in the ticket :)